pyxem/utils/calibration_utils.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2021 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import numpy as np
+from pyxem.utils.indexation_utils import index_dataset_with_template_rotation
+from diffsims.generators.library_generator import DiffractionLibraryGenerator
+
+def find_diffraction_calibration(
+    patterns,calibration_guess,library_phases,lib_gen,size,max_excitation_error = 0.01, **kwargs
+):
+    """Finds the optimal diffraction calibration for a pattern or set of patterns by optimizing correlation scores.
+    
+    Parameters
+    ----------
+    patterns : hyperspy:Signal2D object
+        Diffration patterns to be iteratively matched to find maximum correlation scores.  Should be of known phase.
+    calibration_guess : float
+        Inital value for the diffraction calibration in inverse Angstoms per pixel.
+    library_phases : diffsims:StructureLibrary Object
+        Dictionary of structures and associated orientations for which
+        electron diffraction is to be simulated.  Used to create the DiffractionLibrary.
+    lib_gen : diffsims:DiffractionLibraryGenerator Object
+        Computes a library of electron diffraction patterns for specified atomic
+        structures and orientations.  Used to create the DiffractionLibrary.
+    size : integer
+        How many different steps to test for the first two iterations.
+    max_excitation_error : float
+        Gets passed to get_diffraction_library
+    **kwargs to be passed to index_dataset_with_template_rotation
+    
+    Returns
+    -------
+    mean_cal : float
+        Mean of calibrations found for each pattern.
+    full_corrlines : np.array of shape (size*2 + 20, 2 , number of patterns)
```

Comment:
State the shape in the variable explanation. This line should only be `full_corrlines : numpy.ndarray` for intersphinx to link to NumPy's docs in pyxem's docs.